api/tickets/validate.js

```
@@ -1,87 +1,238 @@
+        isValid: false, 
+        error: 'Token contains invalid characters',
+        securityRisk: true 
+      };
+    }
   }
 
-  if (now > record.resetTime) {
-    rateLimitMap.set(ip, { count: 1, resetTime: now + RATE_LIMIT_WINDOW });
-    return false;
+  // Additional validation for non-printable characters
+  if (/[\x00-\x08\x0B\x0C\x0E-\x1F\x7F-\x9F]/.test(token)) {
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)